File: content/snippets/go/interview_lcci_17_01_add_without_plus.go
package main

import "fmt"

func add(a int, b int) int {
	for b != 0 {
		carry := uint(a&b) << 1
		a = a ^ b
		b = int(carry)
	}
	return a
}

func main() {
	fmt.Println(add(1, 2))   // 3
	fmt.Println(add(3, -2))  // 1
	fmt.Println(add(-1, -2)) // -3
}

File: content/snippets/typescript/interview_lcci_17_01_add_without_plus.ts
function add(a: number, b: number): number {
    while (b !== 0) {
        const carry = (a & b) << 1;
        a = a ^ b;
        b = carry;
    }
    return a;
}

console.log(add(1, 2));    // 3
console.log(add(3, -2));   // 1
console.log(add(-1, -2));  // -3
